src/utils/src/time.rs

```diff
@@ -149,8 +149,7 @@ pub fn get_time_ns(clock_type: ClockType) -> u64 {
     };
     // SAFETY: Safe because the parameters are valid.
     unsafe { libc::clock_gettime(clock_type.into(), &mut time_struct) };
-    seconds_to_nanoseconds(time_struct.tv_sec).expect("Time conversion overflow") as u64
-        + (time_struct.tv_nsec as u64)
+    convert_to_nanoseconds(&time_struct).expect("Time conversion overflow")
```

Comment:
`convert_to_nanoseconds` can be removed and `u64::try_from(...).unwrap()` can be used here.